src/env-inl.h

```diff
@@ -356,6 +356,25 @@ inline IsolateData* Environment::isolate_data() const {
   return isolate_data_;
 }
 
+inline void Environment::context_created(v8_inspector::V8ContextInfo info) {
+  contexts()->push_back(info);
+  if (inspector_agent()->IsStarted()) {
+    inspector_agent()->ContextCreated(info);
```

Comment:
I would suggest to always call ContextCreated/ContextDestroyed - and then let inspector_agent check the IsStarted. This way in this header it will be a single line call (also, contexts vector will live in inspector then).